### PR TITLE
Fix duplicated devDependencies and babel config

### PR DESCRIPTION
**Pull request by jaburcodes**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,9 @@
     "pre-commit": "^1.2.2",
     "prettier": "^1.15.1",
     "typescript": "^3.0.3",
-    "typescript-eslint-parser": "^18.0.0"
+    "typescript-eslint-parser": "^18.0.0",
+    "isomorphic-fetch": "^2.2.1",
```
Comment (PlayMa256) [inline package.json]: why put on the top package, alongside babel-loader? They are not used on more than one package, are they?

Comment (jaburcodes) [reply package.json]: they are

Comment (jaburcodes) [reply package.json]: when using in more than one package, just add to the root.

Comment (jaburcodes) [reply package.json]: monorepo "bons costumes"

Comment (PlayMa256) [reply package.json]: interesting... i already thought the same. @sibelius what do you think? Personally i think having one source of truth and making it easy to update dependencies is a :heavy_plus_sign:

Comment (jaburcodes) [reply package.json]: there are more duplicated deps.. and babel package should be a `babel-preset`.. so u can use in ur babel config.

Check: https://github.com/FotonTech/gigatron

**Comment by sibelius** [comment]
Rethink “best practices”

**Comment by jaburcodes** [comment]
yarn doesnt solve anything for u if you keep duplicated babelrc configs and dependencies.

You are not using full monorepo power.

**Comment by sibelius** [comment]
yarn will nohoist deps for you, if you want to fix babel configs, you need a babel package, also a webpack package, not keeping all of them on root

**Comment by pedronauck** [comment]
> Rethink “best practices”

Really, there's no reason to rethink this practive.
https://github.com/lerna/lerna#common-devdependencies

**Comment by sibelius** [comment]
lerna was the first to tackle monorepo on open source
yarn workspaces came after https://yarnpkg.com/lang/en/docs/workspaces/

yarn workspaces will hoist all this for you automatically, you can avoid nohoist using this option

https://yarnpkg.com/blog/2018/02/15/nohoist/

after yarn pnp we won't need to think about hoist or nohoist anymore, as all the deps will be in a map

**Comment by pedronauck** [comment]
But if you have common devDependency across any packages, why repeat this? There's no logic I thought, even if you use some dependency that's an executable package, you can put it on the root directory and use it inside the package.json scripts. About lerna vs yarn, yeah for sure that lerna came up before yarn, but today lerna it's more like a monorepo cli management than another thing. But about the "common dev dependencies" text, that's is just monorepo best practive, it's not related to lerna att all.

**Pull request closed without merging**